frontend/packages/kubevirt-plugin/src/components/vms/vm-users.tsx

```diff
@@ -0,0 +1,121 @@
+import { VMStatus } from '../../constants/vm/vm-status';
+import { Timestamp } from '@console/internal/components/utils';
+import { isGuestAgentInstalled } from '../dashboards-page/vm-dashboard/vm-alerts';
+
+const tableColumnClasses = [
+  classNames('col-lg-3', 'col-md-3', 'col-sm-4', 'col-xs-6'),
+  classNames('col-lg-3', 'col-md-3', 'col-sm-4', 'hidden-xs'),
+  classNames('col-lg-3', 'col-md-3', 'hidden-sm', 'hidden-xs'),
+  classNames('col-lg-3', 'col-md-3', 'col-sm-4', 'col-xs-6'),
+];
+
+const UsersTableHeader = () => {
+  return [
+    {
+      title: 'User Name',
+      props: { className: tableColumnClasses[0] },
+    },
+    {
+      title: 'Domain',
+      props: { className: tableColumnClasses[1] },
+    },
+    {
+      title: 'Log in time',
+      props: { className: tableColumnClasses[2] },
+    },
+    {
+      title: 'Elapsed logged in time',
+      props: { className: tableColumnClasses[4] },
+    },
+  ];
+};
+UsersTableHeader.displayName = 'UsersTableHeader';
+
+const UsersTableRow = ({ obj, index, key, style }) => {
+  return (
+    <TableRow id={obj?.metadata.uid} index={index} trKey={key} style={style}>
+      <TableData className={tableColumnClasses[0]}>{obj?.metadata?.name}</TableData>
+      <TableData className={classNames(tableColumnClasses[1], 'co-break-word')}>
+        {obj?.metadata?.domain}
+      </TableData>
+      <TableData className={tableColumnClasses[2]}>
+        <Timestamp timestamp={new Date(obj?.metadata?.loginTime).toUTCString()} />
+      </TableData>
+      <TableData className={tableColumnClasses[3]}>{fromNow(obj?.metadata?.loginTime)}</TableData>
+    </TableRow>
+  );
+};
+
+export const VMUsersList = withDashboardResources<VMUsersListProps & DashboardItemProps>(
+  ({ watchURL, stopWatchURL, urlResults, vmi, vmStatusBundle }) => {
+    if (vmStatusBundle.status !== VMStatus.RUNNING) {
+      return <p className="text-muted">Virtual Machine is not running</p>;
+    }
+
+    if (!isGuestAgentInstalled(vmi)) {
+      return (
+        <StatusItem
+          Icon={BlueInfoCircleIcon}
+          message="This VM does not have guest agent installed. Some metrics and management features will not be available."
+        />
+      );
+    }
+
+    const name = getName(vmi);
+    const namespace = getNamespace(vmi);
+    const url = `apis/subresources.kubevirt.io/v1alpha3/namespaces/${namespace}/virtualmachineinstances/${name}/guestosinfo`;
+
+    const usersResult = urlResults.getIn([url, 'data']);
+    const usersResultError = urlResults.getIn([url, 'loadError']);
+
+    const data =
+      usersResult &&
+      usersResult?.userList &&
+      usersResult?.userList.map((u) => ({
+        metadata: {
+          name: u?.userName,
+          domain: u?.domain,
+          loginTime: u?.loginTime && u.loginTime * 1000,
+        },
+      }));
+
+    React.useEffect(() => {
+      watchURL(url);
+      return () => {
+        stopWatchURL(url);
+      };
+    }, [watchURL, stopWatchURL, url]);
```

Comment:
I assume we're going to put this code in a centralized location? I can add it to my existing PR if you want, and maybe create a wrapper class to make it easier to work with. What do you think?

Reply:
@suomiy @yaacov We don't plan on using the dashboard code in the final implementation. We're going to write our own and will likely favor using a hook over a HOC. We're just using the dashboard code as a starting point.

Reply:
+1